core/src/heaviest_subtree_fork_choice.rs

```diff
@@ -906,16 +966,18 @@ impl HeaviestSubtreeForkChoice {
     }
 }
 
-impl TreeDiff for HeaviestSubtreeForkChoice {
+impl<'a> TreeDiff<'a> for &'a HeaviestSubtreeForkChoice {
```

Comment:
Why is the new reference stuff necessary?

Reply:
This is just so we can return an iter (previously we had a slice but now the underlying is a BTreeSet). Could also change it to just return the set as well